WordPress/gutenberg
# Global Styles: Reorganize the global styles UI code base

This PR starts the work to bring some clarity and organization to the global styles UI code base. The end goal being to have a `global-styles` folder that contains "dependency-free" code that can be moved to its own package or to another package... We're not there yet, but this should set us in the right path.

What I did for now:

 - Split each screen to each own `ScreeSomething` component and file: this helps clarify the dependencies of each screen and avoid reluctance on more global variables.
 - Move all the components to a `edit-site/components/global-styles` folder with the entry point being the whole global styles panel (without the connecting bits "Complementary area" that can be specific to each "integration")

Things I want to do but I'm leaving separate:

 - Avoid the global `useGlobalStylesContext` hook and favor more granular hooks.
 - Extract all path computations to a dedicate file (router.js)
 - Move `global-styles-provider` to the `global-styles` folder and make it free of any `edit-site` dependency.

**Testing instructions**

Just make sure the global styles panel still works as expected with no functional change.
